src/azul/tdr.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import re
+from typing import List
+from typing import NamedTuple
+
+from boltons.cacheutils import cachedproperty
+from more_itertools import one
+import requests
+
+from azul.bigquery import GoogleQueryAdapter
+from azul.indexer import BundleFQID
+
+log = logging.getLogger(__name__)
+
+
+class Snapshot(NamedTuple):
+    uuid: str
+    name: str
+
+
+class AzulTDRClient:
+    valid_uuid_re = re.compile(r'[-0-9a-z]*')
```

Comment:
Could be stricter. Then again, the best validation is to actually parse the UUID. Look for sites in our code that raise InvalidUUIDError and extract a utility method for validation of UUIDs.

Since many people erroneously compare UUIDs as strings, which is incorrect because of the case ambiguity, the validation method should also optionally reject UUIDs with upper-case characters in it. Maybe that should even be the default behaviour such that callers need to explicitly opt out to allow upper-case UUIDs.